### PR TITLE
yolo nas demo

### PR DESCRIPTION
**Pull request by agosl**
Add a new demo using Yolo Nas as the object detector

### CODE REVIEW

File: demos/yolo_nas/src/demo.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+import norfair
+from norfair import Detection, Paths, Tracker, Video
+
+DISTANCE_THRESHOLD_BBOX: float = 0.7
+DISTANCE_THRESHOLD_CENTROID: int = 30
+MAX_DISTANCE: int = 10000
+
+
+import super_gradients
```
Comment (javiber) [inline demos/yolo_nas/src/demo.py:15]: imports should be at the top, before the constants

File: demos/yolo_nas/src/demo.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    ) -> torch.tensor:
+
+        if classes is not None:
+            self.model.classes = classes
+
+        detections = self.model.predict(img, iou_threshold, conf_threshold)
+        return detections
+
+
+def yolo_detections_to_norfair_detections(
```
Comment (javiber) [inline demos/yolo_nas/src/demo.py:50]: Not something to implement in this PR but to think about and maybe add to the backlog

We've used this function so often that I believe it should be included in Norfair's utils. Maybe rename it to `xywh_detections_to_norfair` and call it with `yolo_detections.xywh`

File: demos/yolo_nas/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+RUN pip install --upgrade pip
+
+RUN pip install -r https://raw.githubusercontent.com/Deci-AI/super-gradients/master/requirements.txt && \
+    # For some reason it doesn't work with latest version of OpenCV \ --> AttributeError: partially initialized module 'cv2' has no attribute '_registerMatType' (most likely due to a circular import)
+    pip install opencv-python==4.5.5.64
+
+RUN pip install git+https://github.com/tryolabs/norfair.git@master#egg=norfair
+
+RUN pip install super-gradients
```
Comment (javiber) [inline demos/yolo_nas/Dockerfile:17]: we should the requirements from `requirements.txt` or remove the file

File: demos/yolo_nas/requirements.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+super-gradient==3.1.1
```
Comment (javiber) [inline demos/yolo_nas/requirements.txt:1]: see comment above

**Review by javiber — CHANGES_REQUESTED** [review]
Great work! 
Just left some nitpicking and an idea to discuss in the group

**Review by javiber — APPROVED** [review]

**Comment by aguscas** [comment]
Could we maybe add a mention of this demo in the main README.md? It could fit well in the [adding tracking to different detectors](https://github.com/tryolabs/norfair?tab=readme-ov-file#adding-tracking-to-different-detectors) section.

**Comment by aguscas** [comment]
> Could we maybe add a mention of this demo in the main README.md? It could fit well in the [adding tracking to different detectors](https://github.com/tryolabs/norfair?tab=readme-ov-file#adding-tracking-to-different-detectors) section.

I just realize that we need to have It in the master first in order to provide a link to the demo from the main README, so forget whatever I said there.

**Pull request merged**